src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -76,7 +77,14 @@ public static void check(
     final AnnotationHolder holder
   ) {
     final ResultHolder type1 = HaxeTypeResolver.getTypeFromTypeTag(tag, erroredElement);
-    final ResultHolder type2 = HaxeTypeResolver.getPsiElementType(initExpression);
+    ResultHolder t2 = HaxeTypeResolver.getPsiElementType(initExpression);
+
+    ResultHolder aeExpr = HaxeAbstractEnumUtil.getStaticMemberExpression(initExpression.getExpression());
+    if(aeExpr != null) {
+      t2 = aeExpr;
+    }
+    final ResultHolder type2 = t2;
```

Comment:
I'm not asking you to change this.  IMO, it's not worth creating another variable just to get the 'final' designation.  Particularly when it's only going to be used once.